web3/contract.py

```diff
@@ -986,11 +1016,14 @@ def call_contract_function(abi,
     Helper function for interacting with a contract function using the
     `eth_call` API.
     """
+    is_fallback_func = function_name is "fallback"
```

Comment:
FYI, `is` in python means  `is A the exact same object as B` which should not be confused with equality.

```python
In [45]: x = 'b' * 257

In [46]: y = 'b' * 257

In [47]: x is y
Out[47]: False

In [48]: x == y
Out[48]: True
```

Python makes certain optimizations which cause `is` comparisons to work in some cases since python memoizes some values (like integers 1-256)

```python
In [49]: x = 1

In [50]: y = 1

In [51]: x is y
Out[51]: True

In [52]: x == y
Out[52]: True

In [53]: x = 257

In [54]: y = 257

In [55]: x is y
Out[55]: False

In [56]: x == y
Out[56]: True
```

Reply:
Good find, thanks for pointing it out!